File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/CopySample/src/com/hp/workpath/sample/copysample/fragments/ListDialogFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.copysample.fragments;

import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CompoundButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.widget.AppCompatRadioButton;
import androidx.fragment.app.DialogFragment;

import com.hp.workpath.api.copier.StoredJobInfo;
import com.hp.workpath.sample.copysample.R;

import java.util.ArrayList;
import java.util.List;

public class ListDialogFragment extends DialogFragment {

    public static final String CURRENT_SELECTED = "currentSelected";
    public static final String STORED_JOB_INFO_LIST = "storedJobInfoList";

    private TextView mEmptyTextView;
    private RadioGroup mRadioGroup;

    private ArrayList<StoredJobInfo> storedJobInfoList;
    private String currentStoredJobId;

    public static ListDialogFragment newInstance(List<StoredJobInfo> storedJobInfoList,
                                                 String currentStoredJobId) {
        ListDialogFragment f = new ListDialogFragment();
        Bundle args = new Bundle();
        args.putString(CURRENT_SELECTED, currentStoredJobId);
        args.putParcelableArrayList(STORED_JOB_INFO_LIST, new ArrayList<>(storedJobInfoList));
        f.setArguments(args);
        return f;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.dialog_radio_list, null);
        storedJobInfoList = getArguments().getParcelableArrayList(STORED_JOB_INFO_LIST);
        currentStoredJobId = getArguments().getString(CURRENT_SELECTED);
        findViewElements(view);
        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity())
                .setTitle(R.string.stored_job)
                .setView(view)
                .setNegativeButton(android.R.string.cancel, mCancelListener)
                .setCancelable(false);
        return dialogBuilder.create();
    }

    private void findViewElements(View view) {
        mEmptyTextView = view.findViewById(R.id.emptyTextView);
        mRadioGroup = view.findViewById(R.id.radioGroup);

        if (storedJobInfoList.size() == 0) {
            mEmptyTextView.setVisibility(View.VISIBLE);
        } else {
            for (int i = 0; i < storedJobInfoList.size(); i++) {
                final AppCompatRadioButton radio = new AppCompatRadioButton(getActivity());
                RadioGroup.LayoutParams params = new RadioGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
                params.setMargins(0, 5, 0, 5);
                radio.setLayoutParams(params);

                StoredJobInfo storedJobInfo = storedJobInfoList.get(i);
                StringBuilder infoStringBuilder = new StringBuilder();
                infoStringBuilder.append("(").append(storedJobInfo.getStoredJobId()).append(")\n")
                        .append("FolderName: ").append(storedJobInfo.getStoredJobFolderName()).append(", ")
                        .append("JobName: ").append(storedJobInfo.getStoredJobName());
                radio.setText(infoStringBuilder.toString());
                radio.setTag(storedJobInfo.getStoredJobId());
                if (!TextUtils.isEmpty(currentStoredJobId)
                        && currentStoredJobId.equals(storedJobInfo.getStoredJobId())) {
                    radio.setChecked(true);
                }
                radio.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                        Intent intent = new Intent();
                        intent.putExtra(CURRENT_SELECTED, (String) radio.getTag());
                        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, intent);
                        getDialog().dismiss();
                    }
                });
                mRadioGroup.addView(radio);
            }
        }
    }

    private DialogInterface.OnClickListener mCancelListener = new DialogInterface.OnClickListener() {
        @Override
        public void onClick(DialogInterface dialog, int which) {
            dialog.cancel();
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/CopySample/src/com/hp/workpath/sample/copysample/fragments/ListDialogFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.copysample.fragments

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.view.ViewGroup
import android.widget.RadioGroup
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.widget.AppCompatRadioButton
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import com.hp.workpath.api.copier.StoredJobInfo
import com.hp.workpath.sample.copysample.R
import com.hp.workpath.sample.copysample.databinding.DialogRadioListBinding
import com.hp.workpath.sample.copysample.fragments.StoreJobFragment.Companion.REQUEST_KEY

class ListDialogFragment : DialogFragment() {

    private var storedJobInfoList: ArrayList<StoredJobInfo>? = null
    private var currentStoredJobId: String? = null
    private lateinit var mRadioGroup: RadioGroup
    private var mBindingFragment: DialogRadioListBinding? = null

    // This property is only valid between onCreateView and onDestroyView.
    private val mBindingDialogRadioList get() = mBindingFragment!!

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = requireActivity().layoutInflater
        mBindingFragment = DialogRadioListBinding.inflate(inflater)
        val view = mBindingDialogRadioList.root
        storedJobInfoList = requireArguments().getParcelableArrayList(STORED_JOB_INFO_LIST)
        currentStoredJobId = requireArguments().getString(CURRENT_SELECTED)
        findViewElements(view)
        val dialogBuilder = AlertDialog.Builder(requireActivity())
            .setTitle(R.string.stored_job)
            .setView(view)
            .setNegativeButton(android.R.string.cancel, mCancelListener)
            .setCancelable(false)
        return dialogBuilder.create()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBindingFragment = null
    }

    private fun findViewElements(view: View) {
        mRadioGroup = view.findViewById(R.id.radioGroup)
        storedJobInfoList?.let {
            if (it.size == 0) {
                mBindingDialogRadioList.emptyTextView.visibility = View.VISIBLE
            } else {
                for (i in it.indices) {
                    val radio = AppCompatRadioButton(activity)
                    val params = RadioGroup.LayoutParams(
                        ViewGroup.LayoutParams.MATCH_PARENT,
                        ViewGroup.LayoutParams.WRAP_CONTENT
                    )
                    params.setMargins(0, 5, 0, 5)
                    radio.layoutParams = params
                    val storedJobInfo = it[i]
                    val infoStringBuilder = StringBuilder()
                    infoStringBuilder.append("(").append(storedJobInfo.storedJobId).append(")\n")
                        .append("FolderName: ").append(storedJobInfo.storedJobFolderName)
                        .append(", ")
                        .append("JobName: ").append(storedJobInfo.storedJobName)
                    radio.text = infoStringBuilder.toString()
                    radio.tag = storedJobInfo.storedJobId
                    if (!TextUtils.isEmpty(currentStoredJobId)
                        && currentStoredJobId == storedJobInfo.storedJobId
                    ) {
                        radio.isChecked = true
                    }
                    radio.setOnCheckedChangeListener { _, _ ->
                        parentFragmentManager.setFragmentResult(
                            REQUEST_KEY, // Same request key StoreJobFragment used to register its listener
                            bundleOf(CURRENT_SELECTED to radio.tag as String) // The data to be passed to StoreJobFragment
                        )
                        dialog?.dismiss()
                    }
                    mRadioGroup.addView(radio)
                }
            }
        } ?: run {
            mBindingDialogRadioList.emptyTextView.visibility = View.VISIBLE
        }
    }

    private val mCancelListener = DialogInterface.OnClickListener { dialog, _ -> dialog.cancel() }

    companion object {
        const val CURRENT_SELECTED = "currentSelected"
        const val STORED_JOB_INFO_LIST = "storedJobInfoList"

        @JvmStatic
        fun newInstance(
            storedJobInfoList: List<StoredJobInfo>?,
            currentStoredJobId: String?
        ): ListDialogFragment {
            val f = ListDialogFragment()
            val args = Bundle()
            args.putString(CURRENT_SELECTED, currentStoredJobId)
            args.putParcelableArrayList(
                STORED_JOB_INFO_LIST,
                storedJobInfoList?.let { ArrayList(it) })
            f.arguments = args
            return f
        }
    }
}